api/queries_repo.go

```diff
@@ -1023,23 +1092,91 @@ func RepoProjects(client *Client, repo ghrepo.Interface) ([]RepoProject, error)
 	return projects, nil
 }
 
-// RepoAndOrgProjects fetches all open projects for a repository and its org
+// RepoProjectsV2 fetches all open projectsV2 for a repository.
+func RepoProjectsV2(client *Client, repo ghrepo.Interface) ([]RepoProjectV2, error) {
+	type responseData struct {
+		Repository struct {
+			ProjectsV2 struct {
+				Nodes    []RepoProjectV2
+				PageInfo struct {
+					HasNextPage bool
+					EndCursor   string
+				}
+			} `graphql:"projectsV2(first: 100, orderBy: {field: TITLE, direction: ASC}, after: $endCursor)"`
+		} `graphql:"repository(owner: $owner, name: $name)"`
+	}
+
+	variables := map[string]interface{}{
+		"owner":     githubv4.String(repo.RepoOwner()),
+		"name":      githubv4.String(repo.RepoName()),
+		"endCursor": (*githubv4.String)(nil),
+	}
+
+	var projectsV2 []RepoProjectV2
+	for {
+		var query responseData
+		err := client.Query(repo.RepoHost(), "RepositoryProjectV2List", &query, variables)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, p := range query.Repository.ProjectsV2.Nodes {
+			if !p.Closed {
+				projectsV2 = append(projectsV2, p)
+			}
+		}
+
+		if !query.Repository.ProjectsV2.PageInfo.HasNextPage {
+			break
+		}
+		variables["endCursor"] = githubv4.String(query.Repository.ProjectsV2.PageInfo.EndCursor)
+	}
+
+	return projectsV2, nil
+}
+
+// RepoAndOrgProjects fetches all open projects for a repository and its organization.
 func RepoAndOrgProjects(client *Client, repo ghrepo.Interface) ([]RepoProject, error) {
 	projects, err := RepoProjects(client, repo)
 	if err != nil {
-		return projects, fmt.Errorf("error fetching projects: %w", err)
+		return nil, fmt.Errorf("error fetching projects: %w", err)
 	}
 
 	orgProjects, err := OrganizationProjects(client, repo)
-	// TODO: better detection of non-org repos
+	// TODO: Better detection of non-org repos.
 	if err != nil && !strings.Contains(err.Error(), "Could not resolve to an Organization") {
-		return projects, fmt.Errorf("error fetching organization projects: %w", err)
+		return nil, fmt.Errorf("error fetching organization projects: %w", err)
 	}
+
 	projects = append(projects, orgProjects...)
 
 	return projects, nil
 }
 
+// RepoAndOrgProjectsV2 fetches all open projectsV2 for a repository and its organization.
+// Note: If the auth token does not have sufficient scopes or projectsV2 is not supported
+// on the host then those errors are swallowed and nil is returned.
+func RepoAndOrgProjectsV2(client *Client, repo ghrepo.Interface) ([]RepoProjectV2, error) {
+	projectsV2, err := RepoProjectsV2(client, repo)
+	if err != nil {
+		if !strings.Contains(err.Error(), "Your token has not been granted the required scopes to execute this query") ||
+			!strings.Contains(err.Error(), "Field 'ProjectsV2' doesn't exist on type 'Repository'") {
+			return nil, nil
+		}
+
+		return nil, fmt.Errorf("error fetching projectsV2: %w", err)
+	}
+
+	orgProjectsV2, err := OrganizationProjectsV2(client, repo)
```

Comment:
Agreed that we could do that but because of how this function is used in `RepoMetadata` function this change would be require adding some synchronization code around `result.ProjectsV2` in `RepoMetadata` which would add a good amount of complexity for what seems like fairly little payoff. If we notice this is slow or receive complaints then we can optimize this further.